### PR TITLE
[MERGE][FIX] mail: several discuss and messaging menu fixes

### PR DESCRIPTION
**Pull request by alexkuhn**
This merge fixes the following issues

1.  No prefix '#' on DM chats
2. Undated previews appear last in the messaging menu

It also adds a missing test for "[FIX] mail: add mention no set focus on subject" https://github.com/odoo/odoo/commit/6d5df714a7970f69a27296c157b08772ceac5b7d.

Task-ID 1911738
Task-ID 1912001

### CODE REVIEW

File: addons/mail/static/src/js/services/mail_manager.js [hunk 1]
```diff
@@ -1118,13 +1119,14 @@ var MailManager =  AbstractService.extend({
     _sortPreviews: function (previews) {
         var res = previews.sort(function (p1, p2) {
             var unreadDiff = Math.min(1, p2.unreadCounter) - Math.min(1, p1.unreadCounter);
+            var datedDiff = !!p2.date - !!p1.date;
             var isTwoUserThreadDiff = p2.isTwoUserThread - p1.isTwoUserThread;
             var dateDiff = (!!p2.date - !!p1.date) ||
```
Comment (aab-odoo) [inline addons/mail/static/src/js/services/mail_manager.js]: you can remove this part i guess

File: addons/mail/static/tests/discuss_tests.js [hunk 2]
```diff
@@ -423,6 +423,90 @@ QUnit.test('@ mention in channel', function (assert) {
...
+        mockRPC: function (route, args) {
+            if (args.method === 'channel_fetch_listeners') {
+                fetchListenersDef.resolve();
+                return $.when([
+                    {id: 1, name: 'Admin'},
+                ]);
+            }
+            return this._super.apply(this, arguments);
+        },
+        debug: true,
```
Comment (aab-odoo) [inline addons/mail/static/tests/discuss_tests.js]: aie aie aie

File: addons/mail/static/tests/discuss_tests.js [hunk 3]
```diff
@@ -423,6 +423,90 @@ QUnit.test('@ mention in channel', function (assert) {
...
+    })
+    .then(function (discuss) {
+        var $general = discuss.$('.o_mail_discuss_sidebar')
+                        .find('.o_mail_discuss_item[data-thread-id=1]');
+        assert.strictEqual($general.length, 1,
+            "should have the channel item with id 1");
+        assert.strictEqual($general.attr('title'), 'general',
```
Comment (aab-odoo) [inline addons/mail/static/tests/discuss_tests.js]: isn't this assert already done in an other test? This is not really a requirement for your test

**Comment by alexkuhn** [comment]
@aab-odoo Changes made from your comments

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**